### PR TITLE
create interfaces of main entities

### PR DESCRIPTION
**Pull request by hutnikau**

### CODE REVIEW

File: model/SystemStatus/InstanceStatusService.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ * For example, one of TYPE_INSTANCE checks could store hash of config folder for each instance and by
+ * retrieving this hash from SystemStatusLog and comparing it for all instances we may make sure that all instances
+ * have the same configuration
+ *
+ * @package oat\taoSystemStatus\model\SystemStatus
+ * @author Aleh Hutnikau, <hutnikau@1pt.com>
+ */
+class InstanceStatusService extends AbstractSystemStatusService
+{
+    const SERVICE_ID = 'taoSystemStatus/InstanceStatusService';
```
Comment (tikhanovichA) [inline model/SystemStatus/InstanceStatusService.php:42]: Do we need to add a default config for this Service?

File: model/SystemStatus/InstanceStatusService.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    /**
+     * @inheritdoc
+     */
+    public function check(): Report
```
Comment (Yury-Razhkou) [inline model/SystemStatus/InstanceStatusService.php:47]: Current version of phpunit not supporting return type hinting.

File: model/SystemStatus/InstanceStatusService.php (hunk 1 above)
Comment (hutnikau) [reply model/SystemStatus/InstanceStatusService.php:42]: Yes, sure will add later when implementation be ready

File: model/SystemStatus/InstanceStatusService.php (hunk 2 above)
Comment (hutnikau) [reply model/SystemStatus/InstanceStatusService.php:47]: As far as i know we are planning to update phpunit soon.
It breaks only mock builder. Unit test works for me (on v4.8.0)

**Review by tikhanovichA — APPROVED** [review]
Good job!

**Pull request merged**